File: bible-tools/src/main/java/rawcomposition/bibletools/info/model/json/bible/Book.java
package rawcomposition.bibletools.info.model.json.bible;

import com.google.gson.annotations.SerializedName;

import java.io.Serializable;
import java.util.List;

/**
 * Created by tinashe on 2016/01/28.
 */
public class Book implements Serializable {
    private static final long serialVersionUID = 457473970965929827L;

    @SerializedName("title")
    private String title;

    @SerializedName("abr")
    private String abr;

    @SerializedName("chapters")
    private List<Chapter> chapters;

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getAbr() {
        return abr;
    }

    public void setAbr(String abr) {
        this.abr = abr;
    }

    public List<Chapter> getChapters() {
        return chapters;
    }

    public void setChapters(List<Chapter> chapters) {
        this.chapters = chapters;
    }
}

File: bible-tools/src/main/java/rawcomposition/bibletools/info/data/model/Book.kt
package rawcomposition.bibletools.info.data.model

import android.os.Parcel
import android.os.Parcelable
import com.arlib.floatingsearchview.suggestions.model.SearchSuggestion
import com.google.gson.annotations.SerializedName

data class Book(val title: String) : SearchSuggestion {
    override fun getBody(): String {
        return title
    }

    @SerializedName("abr")
    var abr: String? = null

    @SerializedName("chapters")
    var chapters: List<Chapter>? = null

    constructor(parcel: Parcel, title: String) : this(title) {
        abr = parcel.readString()
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(title)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<Book> {
        override fun createFromParcel(parcel: Parcel): Book {
            return Book(parcel.readString())
        }

        override fun newArray(size: Int): Array<Book?> {
            return arrayOfNulls(size)
        }
    }
}
